pkg/cmd/roachtest/run_operation.go

```diff
@@ -0,0 +1,297 @@
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+type opsRunner struct {
+	clusterName string
+	nodeCount   int
+	opsToRun    []registry.OperationSpec
+
+	seed   int64
+	logger *logger.Logger
+
+	datadogEventClient *datadogV1.EventsApi
+	datadogTags        []string
+
+	waitBeforeNextExecution time.Duration
+
+	status struct {
+		syncutil.Mutex
+		// locks out worker from selecting operation to run
+		lockOperationSelection bool
+		// condition variable used to wait for running operation to finish
+		// before running an operation which cannotRunConcurrently
+		operationRunCompleted sync.Cond
+		running               map[string]struct{}
+		lastRun               map[string]time.Time
+	}
+}
+
+// runOperations spins `parallelism` workers to run operations.
+func runOperations(register func(registry.Registry), filter, clusterName string) error {
+	r := makeTestRegistry()
+	register(&r)
+	opSpecs, err := opsToRun(r, filter)
+	if err != nil {
+		return err
+	}
+
+	if err := roachprod.LoadClusters(); err != nil {
+		return err
+	}
+	cluster, ok := roachprod.CachedCluster(clusterName)
+	if !ok {
+		return errors.New("roachtest: run-operation: cluster does not exist")
+	}
+
+	_, seed := randutil.NewTestRand()
+	l, err := logger.RootLogger("", logger.NoTee)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = newDatadogContext(ctx)
+	CtrlC(ctx, l, cancel, nil)
+
+	or := opsRunner{
+		clusterName:             clusterName,
+		nodeCount:               cluster.VMs.Len(),
+		opsToRun:                opSpecs,
+		seed:                    seed,
+		logger:                  l,
+		datadogEventClient:      datadogV1.NewEventsApi(datadog.NewAPIClient(datadog.NewConfiguration())),
+		datadogTags:             getDatadogTags(),
+		waitBeforeNextExecution: roachtestflags.WaitBeforeNextExecution,
+	}
+	or.status.running = make(map[string]struct{})
+	or.status.lastRun = make(map[string]time.Time)
+	or.status.operationRunCompleted.L = &or.status
+
+	var wg errgroup.Group
+	runForever := roachtestflags.RunForever
+	parallelism := int(math.Min(float64(roachtestflags.OperationParallelism), 10))
+	for i := 1; i <= parallelism; i++ {
+		idx := i
+		wg.Go(func() error {
+			or.runWorker(ctx, idx, runForever)
+			return nil
+		})
+	}
+	return wg.Wait()
+}
+
+// runWorker manages the infinite loop for one operation runner worker.
+func (r *opsRunner) runWorker(ctx context.Context, workerIdx int, runForever bool) {
+	rng := rand.New(rand.NewSource(r.seed + int64(workerIdx)))
+	for {
+		if err := ctx.Err(); err != nil {
+			return
+		}
+
+		opSpec := r.selectOperationToRun(ctx, rng, workerIdx)
+		if opSpec == nil {
+			sleepDuration := time.Duration(15+rng.Intn(15)) * time.Second
```

Comment:
Q: Should this be configurable?

If not, Can we make a constant of the numeric value

Reply:
Let the user not worry about such granular configuration of how much time the worker should wait if no operation is selected to run. I have moved the number to a constant for now.